lib/shared/addon/components/container-shell/component.js

```diff
@@ -135,10 +137,30 @@ export default Component.extend(ThrottledResize, {
   },
 
   connect(url) {
+    const intl = get(this, 'intl')
     const socket = new WebSocket(url, 'base64.channel.k8s.io');
 
     set(this, 'socket', socket);
 
+    socket.onclose = (err = {}) => {
+      const term = get(this, 'term')
+
+      get(this, 'growl').fromError(intl.t('containerShell.errors.close', { url: socket.url }))
+
+      if (err.code === 1006) {
```

Comment:
Is there any reason we can't make this more general? We are specifically handling error 1006 but if we check for existence of error alone and then build a generic translation message, then you should be able to pass the error code and error message to the translation so the user is informed of the exact message.

Reply:
cc: @vincent99 do you think we should make this a more general error handler? Are the URL's useful to anyone? we include it in this and the error before but the url is not escaped and looks wonky. If the url is useful we should escape it so its at least readable 

https://www.dropbox.com/s/z14dk05x0ldfqh9/Screenshot%202019-02-20%2010.03.00.png?dl=0